src/tools/rbd_mirror/image_replayer/journal/Replayer.cc

```diff
@@ -23,6 +23,14 @@
 #include "tools/rbd_mirror/image_replayer/journal/ReplayStatusFormatter.h"
 #include "tools/rbd_mirror/image_replayer/journal/StateBuilder.h"
 
+#if defined(WITH_SEASTAR) && !defined(WITH_ALIEN)
```

Comment:
I'm not sure this is needed (or even valid).  The only thing that is built for Seastar is the OSD -- something that the OSD doesn't pull or depend on shouldn't need these ifdefs.  I think rbd-mirror daemon should include `common/perf_counters_collection.h` unconditionally.  @Matan-B please correct me if I'm wrong.

Reply:
I only copied how this include looked everywhere else, e.g.
https://github.com/ceph/ceph/blob/bbcd13bc55ac6c304ecebfcf476c35d1ea699647/src/common/Throttle.h#L18-L22
I strongly dislike copying the whole thing (include preprocessor conditionals) everywhere, and I'd prefer to have one header that does the conditionals, but that's how the Ceph code base looks like.

Reply:
That is common code, meaning that it may be used by the Seastar-ized OSD.  Other daemons and pure client-side code shouldn't be exposed to `WITH_SEASTAR` shenanigans IMO.